File: python/week-08-project-todo/solutions.py
nächste_id = 1


def create_todo(text):
    """Gib ein Dictionary {"id": id, "text": text, "done": False} zurück."""
    ...


def add_todo(todos, text):
    """Gib eine neue Liste mit einem neuen Todo am Ende zurück."""
    ...


def list_todos(todos):
    """Gib eine Liste nur mit den text-Werten zurück."""
    ...


def mark_done(todos, id):
    """Gib eine neue Liste zurück, bei dem das Todo mit der id done=True hat."""
    ...


def remove_todo(todos, id):
    """Gib eine neue Liste ohne das Todo mit der id zurück."""
    ...

File: typescript/week-08-project-todo/solutions.ts
// === Woche 8: Mini Projekt - Todo Liste ===

export interface Todo {
  id: number;
  text: string;
  done: boolean;
}

// Zähler für eindeutige IDs
let nextId: number = 1;

/**
 * Erstellt ein neues Todo-Objekt mit automatischer ID.
 * Setze done standardmäßig auf false.
 */
export function createTodo(text: string): Todo {
  // TODO: Erstelle ein Todo mit id, text und done: false
}

/**
 * Fügt ein neues Todo am Ende des Arrays hinzu.
 */
export function addTodo(todos: Todo[], text: string): Todo[] {
  // TODO: Erstelle ein neues Todo und hänge es an das Array an
}

/**
 * Gibt ein Array mit allen text-Feldern zurück.
 */
export function listTodos(todos: Todo[]): string[] {
  // TODO: Sammle alle text-Felder in einem Array
}

/**
 * Setzt done auf true für das Todo mit der gegebenen id.
 */
export function markDone(todos: Todo[], id: number): Todo[] {
  // TODO: Finde das Todo mit der richtigen id und setze done auf true
}

/**
 * Entfernt das Todo mit der gegebenen id aus dem Array.
 */
export function removeTodo(todos: Todo[], id: number): Todo[] {
  // TODO: Gib ein neues Array ohne das Todo mit der gegebenen id zurück
}

// Export für Tests, damit nextId zurückgesetzt werden kann
export function resetNextId(): void {
  nextId = 1;
}
